packages/kbn-content-management/table_list/src/table_list_view/components/table.tsx

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+/* eslint react-hooks/exhaustive-deps: 2 */
+
+import React, { Dispatch, useCallback } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiBasicTableColumn,
+  EuiButton,
+  EuiInMemoryTable,
+  CriteriaWithPagination,
+  PropertySort,
+} from '@elastic/eui';
+
+import { useServices } from '../services';
+import type { Action } from '../actions';
+import type {
+  State as TableListViewState,
+  Props as TableListViewProps,
+  UserContentCommonSchema,
+} from '../table_list_view';
+
+interface Props<T extends UserContentCommonSchema> {
+  dispatch: Dispatch<Action<T>>;
+  entityName: string;
+  entityNamePlural: string;
+  isFetchingItems: boolean;
+  tableCaption: string;
+  items: TableListViewState<T>['items'];
+  selectedIds: TableListViewState<T>['selectedIds'];
+  searchQuery: TableListViewState<T>['searchQuery'];
+  tableColumns: Array<EuiBasicTableColumn<T>>;
+  tableSort: TableListViewState<T>['tableSort'];
+  pagination: TableListViewState<T>['pagination'];
+  deleteItems: TableListViewProps<T>['deleteItems'];
+}
+
+export function Table<T extends UserContentCommonSchema>({
+  dispatch,
+  items,
+  isFetchingItems,
+  searchQuery,
+  selectedIds,
+  pagination,
+  tableColumns,
+  tableSort,
+  entityName,
+  entityNamePlural,
+  deleteItems,
+  tableCaption,
+}: Props<T>) {
+  const { getSearchBarFilters } = useServices();
+
+  const renderToolsLeft = useCallback(() => {
```

Comment:
This is usually a code smell that this should be its own component.

Reply:
Ok will split up more in a future PR 👍